datalake/backend/tests/test_dataset_versions_endpoint_logging.py

```diff
@@ -0,0 +1,93 @@
+import logging
+from json import dumps
+from unittest import TestCase
+from unittest.mock import MagicMock, patch
+
+from jsonschema import ValidationError  # type: ignore[import]
+from pynamodb.exceptions import DoesNotExist
+from pytest import mark
+
+from ..endpoints.dataset_versions.create import create_dataset_version
+from .utils import Dataset, any_dataset_id, any_s3_url, any_valid_dataset_type
+
+
+class LogTests(TestCase):
+    logger: logging.Logger
+
+    @classmethod
+    def setup_class(cls) -> None:
+        cls.logger = logging.getLogger("datalake.backend.endpoints.dataset_versions.create")
+
+    @mark.infrastructure
+    @patch("datalake.backend.endpoints.dataset_versions.create.sfn_client.start_execution")
+    def test_should_log_payload_and_sfn_response(self, start_execution_mock: MagicMock) -> None:
+        # given
+        dataset_id = any_dataset_id()
+        dataset_type = any_valid_dataset_type()
+        metadata_url = any_s3_url()
+        with Dataset(dataset_id=dataset_id, dataset_type=dataset_type):
+
+            payload = {
+                "httpMethod": "POST",
+                "body": {"metadata-url": metadata_url, "id": dataset_id, "type": dataset_type},
+            }
+
+            start_execution_mock.return_value = sfn_response = {"executionArn": "Some Response"}
+
+            expected_payload_log = dumps({"payload": payload})
+            expected_execution_log = dumps({"response": sfn_response})
+
+            with patch.object(self.logger, "debug") as logger_mock:
+                # when
+                create_dataset_version(payload)
+
+                # then
+                logger_mock.assert_any_call(expected_payload_log)
```

Comment:
If this assertion fails the test doesn't run the second assertion, which can hide another issue. If I understand correctly both assertions require the same setup. If so they should be subtests so that we're sure to run both assertions. If not they should be separate tests.